include/temoto_robot_manager/robot.h

```diff
@@ -46,10 +47,15 @@ class Robot : public temoto_core::BaseSubsystem
   virtual ~Robot();
   void addPlanningGroup(const std::string& planning_group_name);
   void removePlanningGroup(const std::string& planning_group_name);
-  void plan(std::string planning_group_name, geometry_msgs::PoseStamped& target_pose);
-
-  void execute();
-
+  void planManipulationPath(std::string& planning_group_name, const geometry_msgs::PoseStamped& target_pose);
+  void planManipulationPath(std::string& planning_group_name, const std::string& named_target);  
```

Comment:
I still have the same question: is there a reason for the `planning_group_name` argument not being a `const &` ?

Reply:
mmm inside of the function it might change base on this: 
planning_group_name = (planning_group_name == "") ? ftr.getActivePlanningGroup() : planning_group_name;